### PR TITLE
Add one more shim and IVT access for Live Share

### PR DESCRIPTION
**Pull request by jasonmalinowski**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Ick.  LiveShare has to know about an internal detail of Roslyn?  Especially on how to *create* one of these types?  That seems like a very brittle situation.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Yep. We have a detangle plan for the future, but this is trying to fix the past. 😦

**Comment by jinujoseph** [comment]
cc @srivatsn

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi Yep. We have a detangle plan for the future, but this is trying to fix the past. 😦

Sounds good :)

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs [hunk 1]
```diff
@@ -28,6 +28,12 @@ internal partial class InvisibleEditor : IInvisibleEditor
         private OLE.Interop.IOleUndoManager _manager;
         private readonly bool _needsUndoRestored;
 
+        [Obsolete("This is a compatibility shim for Live Share; please do not use it.")]
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs]: 📝 IVTs to external assemblies are use-at-own-risk. Consumers are expected to update code to address changes we make, if and when such changes are made.

**Pull request merged**